### PR TITLE
Temperature gun update 

### PR DESCRIPTION
**Pull request by ToGWtF**
Refer to issue https://github.com/yogstation13/yogstation/issues/445
### Intent of your Pull Request

Adds an area-locked firing pin to the Temperature gun, allowing it to be used easily by Xenobiologists in their lab only without begging for the HoS's approval. Also lets Temperature guns shoot through windows

### CODE REVIEW

**Comment by ShadowDeath6** [comment]
:+1:

**Comment by AsV9** [comment]
Me wants tgstation/-tg-station#16104

**Comment by ToGWtF** [comment]
Gunna work on that. They made it not require a pin, so my work on the xenopin will still be put to good use.

**Comment by ToGWtF** [comment]
todo :~~changelog,~~ ~~inhand sprites~~, ~~fixing the little light thingys that show how much battery is left~~ , further balancing (pretty balanced already)   , ~~adds the item to its own .dm file to be more uniform with other codebases~~

but for now it works yay

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -347,14 +347,170 @@
...
-	icon_state = "freezegun"
-	desc = "A gun that changes temperatures."
+	icon = 'icons/obj/gun_temperatures.dmi'
+	icon_state = "tempgun_4"
+	item_state = "tempgun_4"
+	fire_sound = 'sound/weapons/pulse3.ogg'
+	desc = "A gun that changes the body temperature of its targets."
+	var/temperature = 300
+	var/target_temperature = 300
+	var/e_cost = ""
```
Comment (oisin100) [inline code/modules/projectiles/guns/energy/special.dm]: Why isnt this just `null` or 0? Same goes for the other `= ""` you made. Theres nothing wrong with what you have. But you know... OCD...

**Comment by AsV9** [comment]
Is this ready?

**Comment by ShadowDeath6** [comment]
It uses the stetchkin/syndie pistol inhand, iirc.

**Comment by AsV9** [comment]
The code appears to be using the correct inhand sprites.

**Pull request merged**